Util/Tokenization/src/main/java/org/tribuo/util/tokens/impl/wordpiece/Wordpiece.java

```diff
@@ -0,0 +1,224 @@
+        this.vocab = vocab;
+        this.unknownToken = unknownToken;
+        this.maxInputCharactersPerWord = maxInputCharactersPerWord;
+    }
+
+    public Wordpiece(String vocabPath) {
+        this.vocabPath = vocabPath;
+        try {
+            this.postConfig();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    /**
+     * Initializes an instance of Wordpiece with the given vocabulary, unknown
+     * token, and max word length.
+     * 
+     * @param vocab                     the pre-trained wordpiece vocabulary. See
+     *                                  the contents of e.g.
+     *                                  https://huggingface.co/bert-base-uncased/resolve/main/vocab.txt
+     * @param unknownToken              a string used to indicate a token was not
+     *                                  found in the vocabulary - typically "[UNK]"
+     * @param maxInputCharactersPerWord a maximum to shield against looping over
+     *                                  character-by-character pathologically long
+     *                                  "tokens"
+     */
+    public Wordpiece(String vocabPath, String unknownToken, int maxInputCharactersPerWord) {
+        this.vocabPath = vocabPath;
+        this.unknownToken = unknownToken;
+        this.maxInputCharactersPerWord = maxInputCharactersPerWord;
+        try {
+            this.postConfig();
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public void postConfig() throws IOException {
+        this.vocab = new HashSet<>(IOUtil.getLines(this.vocabPath));
+    }
+
+    /**
+     * A simple whitespace tokenization method that is not used by
+     * {@link WordpieceTokenizer}.
+     * 
+     * @param text the text to tokenize
+     * @return
+     */
+    public static List<String> whitespaceTokenize(String text) {
+        if (text.isEmpty()) {
+            return Collections.emptyList();
+        }
+        return Arrays.asList(text.split("\\s+"));
+    }
+
+    /**
+     * Executes Wordpiece tokenization on the provided text after performing
+     * whitespace tokenization. This method is not called by
+     * {@link WordpieceTokenizer} which calls {@link #wordpiece(String)} directly.
+     * Note that tokens corresponding to word suffixes as indicated in the provided
+     * vocabulary with the sequence "##" prepended to the entry may be returned by
+     * this method. This method does not lowercase the input text or otherwise
+     * modify it in any way.
+     * 
+     * @param text the text to tokenize
+     * @return tokens corresponding to Wordpiece tokenization applied to the input
+     *         text. Some tokens may have a prefix "##" as described above. Some
+     *         tokens may correspond to an unknown token as specified during
+     *         initialization (default "[UNK]")
+     */
+    public List<String> tokenize(String text) {

```

Comment:
Do we still want to have this method given it's now wrapped up another way?

Reply:
removed. fixed in next commit